sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVExprUtils.scala

```diff
@@ -25,13 +25,18 @@ object CSVExprUtils {
    * This is currently being used in CSV reading path and CSV schema inference.
    */
   def filterCommentAndEmpty(iter: Iterator[String], options: CSVOptions): Iterator[String] = {
+    @inline
+    def nonEmptyLine(line: String) : Boolean = {
```

Comment:
should also fix `org.apache.spark.sql.execution.datasources.csv.CSVUtils.filterCommentAndEmpty`